src/main/java/mate/academy/security/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,53 @@
+package mate.academy.security;
+
+import java.util.Optional;
+import mate.academy.exception.AuthenticationException;
+import mate.academy.exception.RegistrationException;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.User;
+import mate.academy.service.UserService;
+import mate.academy.util.HashUtil;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User login(String email, String password) {
+        try {
+            Optional<User> userByEmail = userService.findByEmail(email);
+            if (userByEmail.isEmpty()) {
+                throw new AuthenticationException("Can't authenticate user by this email: "
+                        + userByEmail);
+            }
+            User user = userByEmail.get();
+            String hashedPassword = HashUtil.hashPassword(password, user.getSalt());
+            if (user.getPassword().equals(hashedPassword)) {
+                return user;
+            }
+            throw new AuthenticationException("Wrong password for this email: " + email);
```

Comment:
We should give clue to potential violator which credential is wrong. Please combine both email and pass problems to one which will state that email/pass combination is wrong